tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/MongoDb/MongoDbIntegration.cs

```diff
@@ -101,6 +101,12 @@ internal static Scope CreateScope(object wireProtocol, object connection)
                     {
                         firstElement.TryGetPropertyValue("Name", out operationName);
 
+                        if (operationName == "isMaster" || operationName == "hello")
+                        {
+                            // This is the command used by MongoDb for ping operations
+                            return null;
+                        }
+
```

Comment:
Given this is the only "bail out" point in this method, and that these ping operations are called a lot, is it worth rearranging this method a bit, so we bail out earlier?

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/2216/commits/4314fe9fa66e266277584869b618dbf783faad9e